subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/util/handler/ProcessExceptionHandler.java

```diff
@@ -0,0 +1,27 @@
+package org.shipkit.internal.gradle.util.handler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.gradle.api.GradleException;
+
+public class ProcessExceptionHandler {
```

Comment:
Can we make this simpler?

```java
public class Exceptions {
    public static void handling(Runnable code, Action<Throwable> handler) {
       //...
    }
}
```

Summary:
 - just one class with one static method
 - only one exception handler is needed
 - the handler reuses Action type that comes from Gradle API
 - the handler has just one method, no need for 2 separate "match" and "create"

This way, the code get simpler and the test code should get way simpler! Code review is the last chance to catch complexity before it goes to production!!!

Besides that, I really like the direction of those changes!

Reply:
@mockitoguy Sure, fighting with complexity is a noble cause :) I implemented feedback, I hope you will like it!